python_modules/libraries/dagster-dbt/dagster_dbt/freshness_builder.py

```diff
@@ -0,0 +1,115 @@
+import datetime
+from typing import Sequence
+
+from dagster import (
+    AssetsDefinition,
+)
+from dagster._core.definitions.asset_check_factories.freshness_checks.last_update import (
+    build_last_update_freshness_checks,
+)
+from dagster._core.definitions.asset_check_factories.freshness_checks.time_partition import (
+    build_time_partition_freshness_checks,
+)
+from dagster._core.definitions.asset_check_factories.utils import DEFAULT_FRESHNESS_SEVERITY
+from dagster._core.definitions.asset_check_spec import AssetCheckSeverity
+from dagster._core.definitions.asset_checks import AssetChecksDefinition
+from dagster._core.definitions.time_window_partitions import TimeWindowPartitionsDefinition
+from dagster._core.errors import DagsterInvariantViolationError
+
+from dagster_dbt.utils import ASSET_RESOURCE_TYPES
+
+from .asset_utils import get_manifest_and_translator_from_dbt_assets
+
+
+def build_freshness_checks_from_dbt_assets(
+    dbt_assets: Sequence[AssetsDefinition],
+) -> Sequence[AssetChecksDefinition]:
+    """Returns a sequence of freshness checks constructed from the provided dbt assets.
+
+    Freshness checks can be configured on a per-model basis in the dbt project configuration.
+    For time-partitioned assets, the freshness check configuration mirrors that of the `build_time_partition_freshness_checks` function.
+    As an example:
+    ```yaml
+    models:
+      - name: customers
+        ...
+        meta:
+          dagster:
+            freshness_check:
+              deadline_cron: "0 0 * * *"
+              severity: "WARN" # Optional, defaults to "WARN"
+    ```
+    For non-time-partitioned assets, the freshness check configuration mirrors that of the
+    `build_last_update_freshness_checks` function. `lower_bound_delta` is provided in terms of
+    seconds, and `deadline_cron` is optional.
+    As an example:
+    ```yaml
```

Comment:
We typically do docstring examples like this:
```
    Examples:
        .. code-block:: yaml
```

Reply:
Aware; but wasn't sure if this quite fit that mold. I guess I had some preconception that code examples of that format would fit the same language pattern as the API in question, but perhaps not.

Reply:
> I guess I had some preconception that code examples of that format would fit the same language pattern as the API in question

I don't think that's a prerequisite